qgym/envs/initial_mapping/initial_mapping_state.py

```diff
@@ -64,12 +65,7 @@ def __init__(
         self.steps_done: int = 0
         """Number of steps done since the last reset."""
 
-        fidelity = False  # whether edges include fidelity
-        for _, _, weight in connection_graph.edges.data("weight"):
-            if not isinstance(weight, int):
-                fidelity = True
-                break
-        if fidelity:
+        if has_fidelity(connection_graph):
```

Comment:
Do we want to have the `has_fidelity` check and change the connection parameter `matrix` to have `np.float_` values depending on `has_fidelity`? Or do we want to have that the connection parameter `matrix` always takes `np.float_` values?

Reply:
Yes, we want to check that. I just changed the behavior of the non fidelity case to give us a boolean matrix instead of a box. This is the same as for out `InitialMapping` environment.